src/parallel/MPI_TIMED/mpi_timed.cpp

```diff
@@ -0,0 +1,226 @@
+/**
+ * @file mpi_timed.cpp
+ * @date 11.05.2020
+ * @author JeremyHarisch
+ *
+ * In this file multiple MPI directives are overwritten to enable timing of these methods.
+ * This is possible, because in mpi these methods are using weak symbols.
+ * For more info, google: "MPI profiling interface".
+ */
+
+#include "ProcessTimer.h"
+
+#ifdef ENABLE_TIMED_MPI
+
+int MPI_Pcontrol(const int level, ...) {
+	return ProcessTimer::get().switchProfiling(level);
+}
+
+int MPI_Send(const void *buf, int count, MPI_Datatype datatype, int dest,
+             int tag, MPI_Comm comm) {
+	ProcessTimer::get().startTimer();
```

Comment:
One could also go crazy and make a timer for each type of `MPI` call but I guess that's not what you are after (yet?)